src/hotspot/share/services/heapDumper.cpp

```diff
@@ -1904,16 +2443,23 @@ void VM_HeapDumper::dump_stack_traces() {
   }
 }
 
+// dump the large objects.
+void VM_HeapDumper::dump_large_objects(ObjectClosure* cl) {
+  if (_large_object_list->is_empty()) {
```

Comment:
drain() should be able to handle an empyt list, so this check should not be needed.